From 8dc4ea6e042a0287886ea76177437597736ea0fc Mon Sep 17 00:00:00 2001
From: "Badrappan, Jeevaka" <jeevaka.badrappan@intel.com>
Date: Mon, 19 Jun 2023 19:05:56 +0530
Subject: [PATCH] Fix for wifi not working due to switch from hidl to aidl

Wi-Fi not working as the hwservicemanager is not able to find
android.hardware.wifi@1.0::IWifi/default.

hwservicemanager is not able to find the service as the Wi-Fi service
is switched to aidl and service name has changed too.

Changes done to fix the issue:
- Change the service name to aidl service
- Remove the wifi related VINTF entries from framework_manifest
as those are included as part of respective vendor service binaries.

Tracked-On: OAM-110417
Signed-off-by: Badrappan, Jeevaka <jeevaka.badrappan@intel.com>
---
 .../caas/framework_manifest.xml               | 25 -------------------
 groups/wlan/iwlwifi/product.mk                |  2 +-
 2 files changed, 1 insertion(+), 26 deletions(-)

diff --git a/groups/device-specific/caas/framework_manifest.xml b/groups/device-specific/caas/framework_manifest.xml
index 62366fe..042cfe2 100644
--- a/groups/device-specific/caas/framework_manifest.xml
+++ b/groups/device-specific/caas/framework_manifest.xml
@@ -120,31 +120,6 @@
             <instance>default</instance>
         </interface>
     </hal>
-    <hal format="aidl" optional="true">
-         <name>android.hardware.wifi.supplicant</name>
-         <version>1</version>
-         <interface>
-             <name>ISupplicant</name>
-             <instance>default</instance>
-         </interface>
-    </hal>
-    <hal format="hidl">
-        <name>android.hardware.wifi</name>
-        <transport>hwbinder</transport>
-        <version>1.5</version>
-        <interface>
-            <name>IWifi</name>
-            <instance>default</instance>
-        </interface>
-    </hal>
-    <hal format="aidl" optional="true">
-        <name>android.hardware.wifi.hostapd</name>
-        <version>1</version>
-        <interface>
-            <name>IHostapd</name>
-            <instance>default</instance>
-        </interface>
-    </hal>
     <hal format="hidl">
         <name>android.hardware.media.omx</name>
         <transport>hwbinder</transport>
diff --git a/groups/wlan/iwlwifi/product.mk b/groups/wlan/iwlwifi/product.mk
index 1af7189..f5e0a94 100644
--- a/groups/wlan/iwlwifi/product.mk
+++ b/groups/wlan/iwlwifi/product.mk
@@ -10,7 +10,7 @@ PRODUCT_PACKAGES += \
     TetheringConfigOverlayGsi
 
 PRODUCT_PACKAGES += \
-    android.hardware.wifi@1.0-service
+    android.hardware.wifi-service
 
 #copy iwlwifi wpa config files
 PRODUCT_COPY_FILES += \
-- 
2.40.0

